tools/test/h5copy/CMakeLists.txt

```diff
@@ -4,11 +4,16 @@ project (HDF5_TOOLS_TEST_H5COPY C)
 # --------------------------------------------------------------------
 # Add the h5copy test executables
 # --------------------------------------------------------------------
-if (HDF5_BUILD_GENERATORS AND BUILD_STATIC_LIBS)
+if (HDF5_BUILD_GENERATORS)
   add_executable (h5copygentest ${HDF5_TOOLS_TEST_H5COPY_SOURCE_DIR}/h5copygentest.c)
   target_include_directories (h5copygentest PRIVATE "${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-  TARGET_C_PROPERTIES (h5copygentest STATIC)
-  target_link_libraries (h5copygentest PRIVATE ${HDF5_LIB_TARGET})
+  if (HDF5_BUILD_STATIC_TOOLS)
```

Comment:
One of the existing tools test generator programs, h5watchgentest,  used `HDF5_BUILD_STATIC_TOOLS` to determine whether to build against static or shared. I used the same for the other tools, though it may more more sense just to use `BUILD_STATIC_LIBS`/`BUILD_SHARED_LIBS`.